stumpy/aampi.py

```diff
@@ -247,26 +247,9 @@ def _update_egress(self, t):
         if np.any(~self._T_isfinite[-self._m :]):
             D[:] = np.inf
 
-        core.apply_exclusion_zone(D, D.shape[0] - 1, self._excl_zone, np.inf)
-
-        update_idx = np.argwhere(D < self._P[:, -1]).flatten()
-        for i in update_idx:
-            idx = np.searchsorted(self._P[i], D[i], side="right")
-            core._shift_insert_at_index(self._P[i], idx, D[i])
-            core._shift_insert_at_index(
-                self._I[i], idx, D.shape[0] + self._n_appended - 1
-            )
-            # D.shape[0] is base-1
-
-        # Calculate the (top-k) matrix profile values/indices for the last subsequence
-        # by using its corresponding distance profile `D`
-        self._P[-1] = np.inf
-        self._I[-1] = -1
-        for i, d in enumerate(D):
-            if d < self._P[-1, -1]:
-                idx = np.searchsorted(self._P[-1], d, side="right")
-                core._shift_insert_at_index(self._P[-1], idx, d)
-                core._shift_insert_at_index(self._I[-1], idx, i + self._n_appended)
+        core._update_incremental_PI(
```

Comment:
Maybe it's just me but it feels like `_update_incremental_PI` is doing too much? Maybe we should be splitting into two separate functions rather than one giant one?

Reply:
I feel these two parts are related. This is how I think about it: I say to myself if there is a new data point, and hence a new distance profile that corresponds to the last subsequence, what changes should be made? One, the NN of previous subsgeuqneces might be changed (which is first for-loop) AND the NN of the last subsequence must be determined (the second for-loop). 

Having said that, we might still say that the second for-loop is just finding top-k in a 1D array. So, we can create one function that just does that. But, how about the first for-loop? I am trying to see how that (independent) function should look like. Maybe we just create one function for the second for-loop, `_find_top_k`, and then call it from `_update_incremental_PI`